### ISSUE TITLE
bug in calculate_area

### ISSUE THREAD

**Issue by fsenf** [tobac-project/tobac#333]
The function `calculate_area` takes `features, mask` as argument. If mask is 3D (i.e. incl. time), `calculate_area` fails because internally calculates an area field (`np.array`) which is 2D. Both `mask` and `area` are provided to the`scipy.ndimage` function `labeled_comprehension` which however assumes that the 1st two arguments have the same shape. confict -> 2D != 3D

Here is part of the error
```python
File [~/TROPOS/proj/2022-06-tobac-devel/tobac/tobac/analysis.py:960](https://file+.vscode-resource.vscode-cdn.net/home/senf/TROPOS/proj/2022-06-tobac-devel/tobac/examples/Basics/~/TROPOS/proj/2022-06-tobac-devel/tobac/tobac/analysis.py:960), in calculate_area(features, mask, method_area)
    957 else:
    958     raise ValueError("method undefined")
--> 960 feature_areas = labeled_comprehension(
    961     area, mask.data, features["feature"], np.sum, area.dtype, np.nan
    962 )
    964 features["area"] = feature_areas
    966 return features

File [~/TROPOS/proj/2022-06-tobac-devel/tobac-env-py38/lib/python3.8/site-packages/scipy/ndimage/_measurements.py:511](https://file+.vscode-resource.vscode-cdn.net/home/senf/TROPOS/proj/2022-06-tobac-devel/tobac/examples/Basics/~/TROPOS/proj/2022-06-tobac-devel/tobac-env-py38/lib/python3.8/site-packages/scipy/ndimage/_measurements.py:511), in labeled_comprehension(input, labels, index, func, out_dtype, default, pass_positions)
    509     input, labels = numpy.broadcast_arrays(input, labels)
    510 except ValueError as e:
--> 511     raise ValueError("input and labels must have the same shape "
    512                         "(excepting dimensions with width 1)") from e
    514 if index is None:
    515     if not pass_positions:

ValueError: input and labels must have the same shape (excepting dimensions with width 1)
```

A solution would be to care in the calculations that `area`has always the same shape as `mask`.

As `calculate_area` is not included in the testing I guess it is not really used by the group, right?

**Comment by w-k-jones** [comment]
Good catch, we should probably allow both a 2d area calculation and volume for 3d fields

**Comment by w-k-jones** [comment]
Footprint area of 3D segments is now resolved by #378 . It may be useful to add a function to calculate volume of 3D objects in future, but I will close this for now
